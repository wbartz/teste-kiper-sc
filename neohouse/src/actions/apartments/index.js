import { fetchAPI, LOG } from '../../helpers';

export const APARTMENTS_REQUEST = 'APARTMENTS_REQUEST';
export const APARTMENTS_SUCCESS = 'APARTMENTS_SUCCESS';
export const APARTMENTS_FAILURE = 'APARTMENTS_FAILURE';
export const APARTMENTS_DELETE_REQUEST = 'APARTMENTS_DELETE_REQUEST';
export const APARTMENTS_DELETE_SUCCESS = 'APARTMENTS_DELETE_SUCCESS';
export const APARTMENTS_DELETE_FAILURE = 'APARTMENTS_DELETE_FAILURE';

export const getApartments = (block_id) => async (dispatch) => {
  dispatch({ type: APARTMENTS_REQUEST });

  await fetchAPI(`/blocks/${block_id}`)
    .then(({ data }) => {
      if(data.type === 'success') {
        dispatch({
          type: APARTMENTS_SUCCESS,
          apartments: data.apartments,
        });
        return;
      }
      dispatch({
        type: APARTMENTS_FAILURE,
        error: data.code,
      });
    })
    .catch((error) => {
      LOG(APARTMENTS_FAILURE, error);
      dispatch({
        type: APARTMENTS_FAILURE,
        error,
      });
    });
};

export const removeApartment = (id, onSuccess) => async (dispatch) => {
  dispatch({type: APARTMENTS_DELETE_REQUEST});

  await fetchAPI(`/apartments/${id}`, 'delete')
    .then(({ data }) => {
      if(data.type === 'success') {
        dispatch({
          type: APARTMENTS_DELETE_SUCCESS,
          apartments: data.apartments,
        });
        return onSuccess();
      }
      dispatch({
        type: APARTMENTS_DELETE_FAILURE,
        error: data.code,
      });
    })
    .catch((error) => {
      LOG(APARTMENTS_DELETE_FAILURE, error);
      dispatch({
        type: APARTMENTS_DELETE_FAILURE,
        error,
      });
    });
}
