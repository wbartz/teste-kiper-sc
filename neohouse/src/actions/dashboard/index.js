import { fetchAPI, LOG } from '../../helpers';

export const DASHBOARD_REQUEST = 'DASHBOARD_REQUEST';
export const DASHBOARD_SUCCESS = 'DASHBOARD_SUCCESS';
export const DASHBOARD_FAILURE = 'DASHBOARD_FAILURE';

export const getDashboard = () => async (dispatch) => {
  dispatch({ type: DASHBOARD_REQUEST });

  await fetchAPI('/dashboard')
    .then(({ data }) => {
      if (data.type === 'success') {
        dispatch({
          type: DASHBOARD_SUCCESS,
          dashboard: data.dashboard,
        });
        return;
      }

      LOG(DASHBOARD_FAILURE, data.code);
      dispatch({
        type: DASHBOARD_FAILURE,
        error: data.code,
      });
    })
    .catch((error) => {
      LOG(DASHBOARD_FAILURE, error);
      dispatch({
        type: DASHBOARD_FAILURE,
        error,
      });
    });
};
