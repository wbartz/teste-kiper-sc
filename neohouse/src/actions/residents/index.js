import { fetchAPI, LOG } from '../../helpers';

export const RESIDENTS_REQUEST = 'RESIDENTS_REQUEST';
export const RESIDENTS_SUCCESS = 'RESIDENTS_SUCCESS';
export const RESIDENTS_FAILURE = 'RESIDENTS_FAILURE';
export const RESIDENTS_DELETE_REQUEST = 'RESIDENTS_DELETE_REQUEST';
export const RESIDENTS_DELETE_SUCCESS = 'RESIDENTS_DELETE_SUCCESS';
export const RESIDENTS_DELETE_FAILURE = 'RESIDENTS_DELETE_FAILURE';

export const getResidents = (apartment_id) => async (dispatch) => {
  dispatch({ type: RESIDENTS_REQUEST });

  await fetchAPI(`/residents/${apartment_id}`)
    .then(({ data }) => {
      if(data.type === 'success') {
        dispatch({
          type: RESIDENTS_SUCCESS,
          residents: data.residents,
        });
        return;
      }
      dispatch({
        type: RESIDENTS_FAILURE,
        error: data.code,
      });
    })
    .catch((error) => {
      LOG(RESIDENTS_FAILURE, error);
      dispatch({
        type: RESIDENTS_FAILURE,
        error,
      });
    });
};

export const removeApartment = (id, onSuccess) => async (dispatch) => {
  dispatch({type: RESIDENTS_DELETE_REQUEST});

  await fetchAPI(`/residents/${id}`, 'delete')
    .then(({ data }) => {
      if(data.type === 'success') {
        dispatch({
          type: RESIDENTS_DELETE_SUCCESS,
          apartments: data.apartments,
        });
        return onSuccess();
      }
      dispatch({
        type: RESIDENTS_DELETE_FAILURE,
        error: data.code,
      });
    })
    .catch((error) => {
      LOG(RESIDENTS_DELETE_FAILURE, error);
      dispatch({
        type: RESIDENTS_DELETE_FAILURE,
        error,
      });
    });
}
