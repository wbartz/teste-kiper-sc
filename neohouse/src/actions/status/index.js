export const RESET_ERROR_MESSAGES = 'RESET_ERROR_MESSAGES';

export const clearErrors = () => dispatch => {
  dispatch({ type: RESET_ERROR_MESSAGES });
};
