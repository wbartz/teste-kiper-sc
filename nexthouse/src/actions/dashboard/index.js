import { fetchAPI, LOG } from '../../helpers';

export const DASHBOARD_REQUEST = 'DASHBOARD_REQUEST';
export const DASHBOARD_SUCCESS = 'DASHBOARD_SUCCESS';
export const DASHBOARD_FAILURE = 'DASHBOARD_FAILURE';

export const getDashboard = (onSuccess) => async (dispatch) => {
  dispatch({ type: DASHBOARD_REQUEST });

  await fetchAPI('/dashboard')
    .then((result) => {
      const { data } = result;
      dispatch({
        type: DASHBOARD_SUCCESS,
        dashboard: data,
      });
      return onSuccess(data);
    })
    .catch((error) => {
      LOG(DASHBOARD_FAILURE, error);
      dispatch({
        type: DASHBOARD_FAILURE,
        error,
      });
    });
};
