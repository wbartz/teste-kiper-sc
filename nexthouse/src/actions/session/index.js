import { fetchAPI, LOG, removeCookie, setCookie } from '../../helpers';

export const EMAIL_SIGNIN_REQUEST = 'EMAIL_SIGNIN_REQUEST';
export const EMAIL_SIGNIN_SUCCESS = 'EMAIL_SIGNIN_SUCCESS';
export const EMAIL_SIGNIN_FAILURE = 'EMAIL_SIGNIN_FAILURE';
export const SIGNOUT_REQUEST = 'SIGNOUT_REQUEST';
export const SIGNOUT_SUCCESS = 'SIGNOUT_SUCCESS';
export const SIGNOUT_FAILURE = 'SIGNOUT_FAILURE';

export const signOut = (onSuccess) => (dispatch) => {
  dispatch({ type: EMAIL_SIGNIN_REQUEST });
  removeCookie('token');
  dispatch({ type: SIGNOUT_SUCCESS });
  return onSuccess();
};

export const signIn = (username, password, onSuccess) => async (dispatch) => {
  dispatch({ type: EMAIL_SIGNIN_REQUEST });

  await fetchAPI('/login', 'post', { username, password })
    .then((result) => {
      const { data } = result;
      setCookie('token', data.token);
      setCookie('user', data.user);
      dispatch({
        type: EMAIL_SIGNIN_SUCCESS,
        token: data.token,
      });
      return onSuccess();
    })
    .catch((error) => {
      LOG(EMAIL_SIGNIN_FAILURE, error.message);
      dispatch({
        type: EMAIL_SIGNIN_FAILURE,
        error: error.message,
      });
    });
};
