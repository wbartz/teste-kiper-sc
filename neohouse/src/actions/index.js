import * as statusActions from './status';
import * as sessionActions from './session';
import * as dashboardActions from './dashboard';
import * as apartmentActions from './apartments';
import * as residentsActions from './residents';

export default {
  ...statusActions,
  ...sessionActions,
  ...dashboardActions,
  ...apartmentActions,
  ...residentsActions,
};
