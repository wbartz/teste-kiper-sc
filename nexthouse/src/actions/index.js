import * as statusActions from './status';
import * as sessionActions from './session';

export default {
  ...statusActions,
  ...sessionActions,
};
