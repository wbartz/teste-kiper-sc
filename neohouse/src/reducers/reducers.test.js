import AppReducer from '.';
import {
  EMAIL_SIGNIN_FAILURE,
  EMAIL_SIGNIN_REQUEST,
  EMAIL_SIGNIN_SUCCESS,
  SIGNOUT_FAILURE,
  SIGNOUT_REQUEST,
  SIGNOUT_SUCCESS,
} from '../actions/session';
import { RESET_ERROR_MESSAGES } from '../actions/status';
import {
  DASHBOARD_REQUEST,
  DASHBOARD_SUCCESS,
  DASHBOARD_FAILURE,
} from '../actions/dashboard';
import {
  APARTMENTS_REQUEST,
  APARTMENTS_SUCCESS,
  APARTMENTS_FAILURE,
  APARTMENTS_DELETE_REQUEST,
  APARTMENTS_DELETE_SUCCESS,
  APARTMENTS_DELETE_FAILURE,
} from '../actions/apartments';
import {
  RESIDENTS_REQUEST,
  RESIDENTS_SUCCESS,
  RESIDENTS_FAILURE,
  RESIDENTS_DELETE_REQUEST,
  RESIDENTS_DELETE_SUCCESS,
  RESIDENTS_DELETE_FAILURE,
} from '../actions/residents';

describe('Status Actions', () => {
  it('faultback to a initial state', () => {
    expect(AppReducer(undefined, { type: null })).toMatchSnapshot();
  });

  it('Clear message errors', () => {
    const request = AppReducer(undefined, { type: RESET_ERROR_MESSAGES });

    expect(request).toHaveProperty('error', null);
  });
});

describe('Session Actions', () => {
  it('faultback to a initial state', () => {
    expect(AppReducer(undefined, { type: null })).toMatchSnapshot();
  });

  it('signIn', () => {
    const error = 'Ocorreu um erro durante a requisição. Por favor tente novamente!';

    const request = AppReducer(undefined, { type: EMAIL_SIGNIN_REQUEST });
    const success = AppReducer(undefined, {
      type: EMAIL_SIGNIN_SUCCESS,
      token: 'abc',
    });
    const failure = AppReducer(undefined, {
      type: EMAIL_SIGNIN_FAILURE,
      error,
    });

    expect(request).toHaveProperty('isLoading', true);

    expect(success).toHaveProperty('isLoading', false);
    expect(success).toHaveProperty('token', 'abc');

    expect(failure).toHaveProperty('isLoading', false);
    expect(failure).toHaveProperty('error', error);
  });

  it('signOut', () => {
    const error = 'Ocorreu um erro durante a requisição. Por favor tente novamente!';

    const request = AppReducer(undefined, { type: SIGNOUT_REQUEST });
    const success = AppReducer(undefined, { type: SIGNOUT_SUCCESS });
    const failure = AppReducer(undefined, { type: SIGNOUT_FAILURE, error });

    expect(request).toHaveProperty('isLoading', true);

    expect(success).toHaveProperty('token', null);

    expect(failure).toHaveProperty('isLoading', false);
    expect(failure).toHaveProperty('error', error);
  });
});

describe('Dashboard actions', () => {
  it('faultback to a initial state', () => {
    expect(AppReducer(undefined, { type: null })).toMatchSnapshot();
  });

  it('getDashboard', () => {
    const error = 'fail';
    const request = AppReducer(undefined, { type: DASHBOARD_REQUEST });
    const success = AppReducer(undefined, {
      type: DASHBOARD_SUCCESS,
      dashboard: [],
    });
    const failure = AppReducer(undefined, {
      type: DASHBOARD_FAILURE,
      error,
    });

    expect(request).toHaveProperty('isLoading', true);

    expect(success).toHaveProperty('isLoading', false);
    expect(success).toHaveProperty('dashboard', []);

    expect(failure).toHaveProperty('isLoading', false);
    expect(failure).toHaveProperty('error', error);
  });
});
