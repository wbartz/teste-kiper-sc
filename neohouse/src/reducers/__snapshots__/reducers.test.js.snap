// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard actions faultback to a initial state 1`] = `
Object {
  "dashboard": Array [],
  "error": null,
  "isLoading": false,
  "token": null,
  "user": null,
}
`;

exports[`Session Actions faultback to a initial state 1`] = `
Object {
  "dashboard": Array [],
  "error": null,
  "isLoading": false,
  "token": null,
  "user": null,
}
`;

exports[`Status Actions faultback to a initial state 1`] = `
Object {
  "dashboard": Array [],
  "error": null,
  "isLoading": false,
  "token": null,
  "user": null,
}
`;
